### PR TITLE
api doc: add document for /api/create-package

### PR DESCRIPTION
**Pull request by zqfan**
fix: https://github.com/composer/packagist/issues/1119

### CODE REVIEW

**Comment by zqfan** [comment]
failed to launch dev env because
```
Problem 1
    - Installation request for ocramius/package-versions 1.11.0 -> satisfiable by ocramius/package-versions[1.11.0].
    - ocramius/package-versions 1.11.0 requires composer-plugin-api ^2.0.0 -> no matching package found.
```
failed to update `composer update` because
```
PHP Fatal error:  Allowed memory size of 1610612736 bytes exhausted
```

SO PLEASE AWARE THIS PR IS NOT VERIFIED LOCALLY!!

**Comment by stof** [comment]
> failed to launch dev env because

this can be solved by running `composer install` with Composer 2

File: src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig [hunk 1]
```diff
@@ -259,5 +260,22 @@ GET https://{{ packagist_host }}/api/security-advisories/?updatedSince=[timestam
 
 </section>
 
+<section class="col-d-12">
+  <h3 id="create-package">{{ 'api_doc.api_create_package'|trans }}</h3>
+
+  <h4 id="create-package-by-url">{{ 'api_doc.api_create_package'|trans }}</h4>
+
+    <p>This interface creates a package for a specific repo. Parameters <code>username</code> and <code>apiToken</code> are required. Only <code>POST</code> method is allowed.</p>
```
Comment (stof) [inline src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig:268]: should be `This endpoint`, not `This interface`

File: src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig [hunk 2]
```diff
@@ -259,5 +260,22 @@ GET https://{{ packagist_host }}/api/security-advisories/?updatedSince=[timestam
...
+
+    <pre>
+POST https://{{ packagist_host }}/api/create-package?username=[username]&apiToken=[apiToken] -d '{"repository":{"url":"[url]"}}'
+<code>
+{
+  "status": "success"
+}
+</code></pre>
+    <p>Working example: <code>curl -X POST 'https://{{ packagist_host }}/api/create-package?username=zqfan&apiToken=********' -d '{"repository":{"url":"https://github.com/tencentcloud/tencentcloud-sdk-php"}}'</code></p>
```
Comment (stof) [inline src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig:277]: I suggest using the repo of `monolog/monolog` to be consistent with other examples

File: src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig [hunk 3]
```diff
@@ -259,5 +260,22 @@ GET https://{{ packagist_host }}/api/security-advisories/?updatedSince=[timestam
...
 
+<section class="col-d-12">
+  <h3 id="create-package">{{ 'api_doc.api_create_package'|trans }}</h3>
+
+  <h4 id="create-package-by-url">{{ 'api_doc.api_create_package'|trans }}</h4>
+
+    <p>This interface creates a package for a specific repo. Parameters <code>username</code> and <code>apiToken</code> are required. Only <code>POST</code> method is allowed.</p>
+
+    <pre>
+POST https://{{ packagist_host }}/api/create-package?username=[username]&apiToken=[apiToken] -d '{"repository":{"url":"[url]"}}'
```
Comment (stof) [inline src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig:271]: you should escape the `&` into `&amp;` to have valid HTML (same below)

File: src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig (hunk 1 above)
Comment (zqfan) [reply src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig:268]: OK

**Comment by zqfan** [comment]
> > failed to launch dev env because
> 
> this can be solved by running `composer install` with Composer 2

Yes, composer2 does fix the problem, although there are some other issues, but finally I can run the dev env.
Thanks a lot!

File: src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig (hunk 2 above)
Comment (zqfan) [reply src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig:277]: sure

File: src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig (hunk 3 above)
Comment (zqfan) [reply src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig:271]: they are in <pre> block, and the examples above in the document don't escape as well, it seems ok to remain plain text.

**Comment by zqfan** [comment]
screeshot of this pr:
image

File: src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig (hunk 3 above)
Comment (stof) [reply src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig:271]: depends which example. Some of them are missing escaping but not all.

And `<pre>` does not mean that the content is not HTML markup anymore.

Comment (zqfan) [reply src/Packagist/WebBundle/Resources/views/api_doc/index.html.twig:271]: done

**Review by stof — APPROVED** [review]

**Comment by Seldaek** [comment]
Thanks

**Pull request merged**